CRM/PivotData/DataLeave.php

```diff
@@ -0,0 +1,566 @@
+    return !empty($this->standardHoursOptions[$locationHoursID]) ?
+      $this->standardHoursOptions[$locationHoursID] : '';
+  }
+
+  /**
+   * Function to load some data on instantiation of this class.
+   */
+  private function initData() {
+    $this->standardHoursOptions = $this->getStandardHoursOptions();
+  }
+
+  /**
+   * Gets options for the location_standard_hours handler.
+   *
+   * @return array
+   */
+  private function getStandardHoursOptions() {
+    $result = civicrm_api3('HRHoursLocation', 'get');
+    $options = [];
+
+    foreach($result['values'] as $value) {
+      $options[$value['id']] = $value['location'] .
+        ' (' . $value['standard_hours'] .
+        ' / ' . $value['periodicity'] . ')';
+    }
+
+    return $options;
+  }
+
+  /**
+   * Get Pay Scale options needed for the pay_scale field.
+   *
+   * @return array
+   */
+  private function getPayScales() {
+    $result = civicrm_api3('HRPayScale', 'get');
+
+    return array_column($result['values'], 'pay_scale', 'id');
+  }
+
+  /**
+   * Gets the Age from the given birth date.
+   *
+   * @param \DateTime $birthDate
+   *
+   * @return int
+   */
+  private function getAge(DateTime $birthDate) {
+    $age = 0;
+    if ($birthDate < new DateTime('today')) {
+      $ageData = CRM_Utils_Date::calculateAge($birthDate->format('Y-m-d'));
+      $ageInYears = CRM_Utils_Array::value('years', $ageData);
+
+      if (isset($ageInYears)) {
+        $age = $ageInYears;
+      }
+    }
+
+    return $age;
+  }
+
+  /**
+   * Handler for the location_standard_hours field.
+   *
+   * @param string $birthDate
+   * @param array $rowData
+   *   Data for other fields belonging to same row as this field value.
+   *
+   * @return string
+   */
+  protected function birthDateHandler($birthDate, $rowData) {
+    if ($birthDate) {
+      return 'Not Set';
+    }
+
+    return $this->getAge(new DateTime($birthDate));
+  }
+
+
+  /**
+   * Handler for the length_of_service Contact custom field.
+   *
+   * @param int $lengthOfService
+   * @param array $rowData
+   *   Data for other fields belonging to same row as this field value.
+   *
+   * @return string
+   */
+  protected function lengthOfServiceHandler($lengthOfService, $rowData) {
+    if (empty($lengthOfService)) {
+      return 'Not Set';
+    }
+
+    $length = [];
+    $today = new DateTime();
+    $past = (new DateTime())->sub(new DateInterval('P' . $lengthOfService . 'D'));
+    $interval = $today->diff($past);
+
+    $years = intval($interval->format('%y'));
+    $months = intval($interval->format('%m'));
+    $days = intval($interval->format('%d'));
+
+    if ($years > 0) {
+      $length[] = $years > 1 ? "$years years" : "$years year";
+    }
+    if ($months > 0) {
+      $length[] = $months > 1 ? "$months months" : "$months month";
+    }
+    if ($days > 0) {
+      $length[] = $days > 1 ? "$days days" : "$days day";
+    }
+
+    return implode(' ', $length);
```

Comment:
If I'm not mistaken, we already have this logic somewhere in CiviHR. Once we actually implement this in CiviHR and not here, it would be a good idea to use the logic we already have there instead of duplicating it.

Reply:
I copied the logic over from the employee portal. The logic resides in the views handler for the Age field for the Leave Reports view. Not sure how we can re-use this, we might need to copy over to CiviHR I think.

Reply:
Since the idea is to replace the existing views-based Leave Report with this extension, I think that all we'll need to do is to put this logic in a separate class so that we can use it in the new report and in anywhere else where we might need it in the future.